vulnerabilities/importers/debian.py

```diff
@@ -21,115 +21,197 @@
 #  VulnerableCode is a free software tool from nexB Inc. and others.
 #  Visit https://github.com/nexB/vulnerablecode/ for support and download.
 
-import dataclasses
+import logging
 from typing import Any
+from typing import Iterable
 from typing import List
 from typing import Mapping
-from typing import Set
 
 import requests
-from dateutil import parser as dateparser
+from django.db.models.query import QuerySet
 from packageurl import PackageURL
+from univers.version_range import DebianVersionRange
+from univers.versions import DebianVersion
 
+from vulnerabilities.helpers import AffectedPackage as LegacyAffectedPackage
+from vulnerabilities.helpers import get_item
+from vulnerabilities.helpers import get_unique_patched_packages_with_affected_packages
 from vulnerabilities.helpers import nearest_patched_package
 from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
 from vulnerabilities.importer import Importer
 from vulnerabilities.importer import Reference
+from vulnerabilities.importer import UnMergeablePackageError
+from vulnerabilities.improver import MAX_CONFIDENCE
+from vulnerabilities.improver import Improver
+from vulnerabilities.improver import Inference
+from vulnerabilities.models import Advisory
 
+logger = logging.getLogger(__name__)
 
-class DebianImporter(Importer):
-    def __enter__(self):
-        if self.response_is_new():
-            self._api_response = self._fetch()
+IGNORABLE_VERSIONS = {"3.8.20-4."}
 
-        else:
-            self._api_response = {}
 
-    def updated_advisories(self) -> Set[AdvisoryData]:
-        advisories = []
+class DebianImporter(Importer):
 
-        for pkg_name, records in self._api_response.items():
-            advisories.extend(self._parse(pkg_name, records))
+    spdx_license_expression = "MIT"
```

Comment:
IMHO we should ask Debian folks  what the license is for https://security-tracker.debian.org/tracker/

Reply:
FYI:

> #debian-security
> [2022-05-11 07:23:53] → Joined channel #debian-security
> <pombreda> Howdy :)
> <pombreda> I wonder what is the license of Debian security data https://github.com/nexB/vulnerablecode/pull/723#discussion_r869695559
> <pombreda> (I need to know as we integrate this in VulnerableCode which is a FOSS db of FOSS vulnerability data)
> <pombreda> carnil, jmm: would you know?

> <jmm_> pombreda: good question! I'm pretty sure the individual bits are hardly copyrighteable (we include e.g. CVE IDs which pretty surely don't have a license either)
> <jmm_> I rather see it as run time state, but if for some paper trail you need some sign off to include the data in VulnerableCode please send a mail to team@security.debian.org and we can ack it

> <pombreda> jmm_: thanks. And I want to ensure that we have a clear tracking.
> jmm_: FWIW, a CVE id proper does not have a license, but the CVE database has some licensing. Mitre's is at https://cve.mitre.org/about/termsofuse.html  and the NIST is a US govt. public domain https://nvd.nist.gov/general/FAQ-Sections/General-FAQs#faqLink7 
>

Reply:
Can you paste the email you send here?
And when you have a reply, we will need the email and reply in a comment and in a .ABOUT file side-by-side

Reply:
From: Tushar Goel <tgoel@nexb.com>
    Date: Thu, May 12, 2022 at 11:42 PM +00:00
    Subject: Usage of Debian Security Data in VulnerableCode
    To: <team@security.debian.org>

    Hey,

    We would like to integrate the debian security data in vulnerablecode
    [1][2] which is a FOSS db of FOSS vulnerability data. We were not able
    to know under which license the debian security data comes. We would
    be grateful to have your acknowledgement over usage of the debian
    security data in vulnerablecode and have some kind of licensing
    declaration from your side.

    [1] - https://github.com/nexB/vulnerablecode
    [2] - https://github.com/nexB/vulnerablecode/pull/723

Reply:
From: Moritz Mühlenhoff <[jmm@inutil.org](mailto:jmm@inutil.org)>
Date: Wed, May 18, 2022, 12:42 AM
Subject: Re: Usage of Debian Security Data in VulnerableCode
To: Tushar Goel <[tgoel@nexb.com](mailto:tgoel@nexb.com)>
Cc: <[team@security.debian.org](mailto:team@security.debian.org)>


Am Thu, May 12, 2022 at 05:12:48PM +0530 schrieb Tushar Goel:
> Hey,
>
> We would like to integrate the debian security data in vulnerablecode
> [1][2] which is a FOSS db of FOSS vulnerability data. We were not able
> to know under which license the debian security data comes. We would
> be grateful to have your acknowledgement over usage of the debian
> security data in vulnerablecode and have some kind of licensing
> declaration from your side.

We don't have a specific license, but you have our endorsemen to
reuse the data by all means :-)

Cheers,
        Moritz